ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -359,24 +406,41 @@ export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
           return (
             <Box sx={{ height, width }}>
               {y_axis && y_axis.show && y_axis.label && <YAxisLabel name={y_axis.label} height={height} />}
-              <LineChart
-                ref={lineChartRef}
-                height={height}
-                data={graphData}
-                yAxis={echartsYAxis}
-                unit={unit}
-                grid={gridOverrides}
-                tooltipConfig={{ wrapLabels: true }}
-                syncGroup="default-panel-group" // TODO: make configurable from dashboard settings and per panel-group overrides
-                onDataZoom={handleDataZoom}
-                //  Show an empty chart when there is no data because the user unselected all items in
-                // the legend. Otherwise, show a "no data" message.
-                noDataVariant={
-                  !graphData.timeSeries.length && graphData.legendItems && graphData.legendItems.length > 0
-                    ? 'chart'
-                    : 'message'
-                }
-              />
+              {showLegacyChart ? (
+                <LineChart
+                  ref={lineChartRef}
+                  height={height}
+                  data={graphData}
+                  yAxis={echartsYAxis}
+                  unit={unit}
+                  grid={gridOverrides}
+                  tooltipConfig={{ wrapLabels: true }}
+                  syncGroup="default-panel-group" // TODO: make configurable from dashboard settings and per panel-group overrides
+                  onDataZoom={handleDataZoom}
+                  //  Show an empty chart when there is no data because the user unselected all items in
+                  // the legend. Otherwise, show a "no data" message.
+                  noDataVariant={
+                    !graphData.timeSeries.length && legendItems && legendItems.length > 0 ? 'chart' : 'message'
+                  }
+                />
+              ) : (
+                <TimeChart
+                  ref={lineChartRef}
```

Comment:
Should we call this something like `chartRef` since it can be used by both LineChart and TimeChart?